Library/OcTextInputLib/OcTextInputLib.c

```diff
@@ -0,0 +1,526 @@
+    }
+
+    // For EFI 1.1 compatibility, do not set shift state
+    // This ensures compatibility with MenuBarDispatchControlHotKey which has
+    // two different code paths for CTRL handling.
+    if ((KeyData->Key.UnicodeChar >= 0x01) && (KeyData->Key.UnicodeChar <= 0x1A)) {
+      DEBUG (
+        (DEBUG_VERBOSE, "OcTextInputLib: CTRL+%c detected (0x%02X), preserving EFI 1.1 compatibility (no shift state)\n",
+         (CHAR8)('A' + KeyData->Key.UnicodeChar - 1), KeyData->Key.UnicodeChar)
+        );
+    }
+
+    // Always keep shift and toggle states at 0 for EFI 1.1 compatibility
+    KeyData->KeyState.KeyShiftState  = 0;
+    KeyData->KeyState.KeyToggleState = 0;
+  }
+
+  return Status;
+}
+
+EFI_STATUS
+EFIAPI
+OcCompatReadKeyStrokeEx (
+  IN EFI_SIMPLE_TEXT_INPUT_EX_PROTOCOL  *This,
+  OUT EFI_KEY_DATA                      *KeyData
+  )
+{
+  // Use the shared implementation
+  return OcReadKeyStrokeFromTextInput (KeyData);
+}
+
+/**
+   Compatibility implementation of SetState (does nothing on EFI 1.1).
+
+   @param  This                 Protocol instance pointer.
+   @param  KeyToggleState       A pointer to the EFI_KEY_TOGGLE_STATE to set the
+                               state for the input device.
+
+   @retval EFI_SUCCESS          The device state was set successfully.
+   @retval EFI_UNSUPPORTED      The device does not support the ability to have its state set.
+ **/
+EFI_STATUS
+EFIAPI
+OcCompatSetState (
+  IN EFI_SIMPLE_TEXT_INPUT_EX_PROTOCOL  *This,
+  IN EFI_KEY_TOGGLE_STATE               *KeyToggleState
+  )
+{
+  // Always return success to prevent calling applications from failing
+  // EFI 1.1 systems don't support toggle state, but we pretend we do
+  // to maintain compatibility with applications that expect this to work
+  return EFI_SUCCESS;
+}/**
+   Compatibility implementation of RegisterKeyNotify (does nothing on EFI 1.1).
+
+   @param  This                 Protocol instance pointer.
+   @param  KeyData              A pointer to a buffer that defines the keystroke
+                               information for the notification function.
+   @param  KeyNotificationFunction Points to the function to be called when the key
+                               sequence is typed in the specified input device.
+   @param  NotifyHandle         Points to the unique handle assigned to the registered notification.
+
+   @retval EFI_SUCCESS          The notification function was registered successfully.
+   @retval EFI_OUT_OF_RESOURCES Failed to allocate memory for the dummy handle.
+ **/
+
+EFI_STATUS
+EFIAPI
+OcCompatRegisterKeyNotify (
+  IN EFI_SIMPLE_TEXT_INPUT_EX_PROTOCOL  *This,
+  IN EFI_KEY_DATA                       *KeyData,
+  IN EFI_KEY_NOTIFY_FUNCTION            KeyNotificationFunction,
+  OUT VOID                              **NotifyHandle
+  )
+{
+  // Not supported on EFI 1.1 systems, but return success with unique handle
+  // to avoid breaking applications that expect this to work
```

Comment:
Do we actually have a known reason for providing this?

Unless `edit` (or something else we know) won't work without this dummy implementation, IMO it would be better to actually return an error code and not seem to support this when we really don't.